Clarified a few points about DATABASES.TIME_ZONE (#17444)
django/django: docs/ref/settings.txt

> When :setting:`USE_TZ` is ``True`` and this option is set, reading datetimes
> from the database returns aware datetimes in this time zone instead of UTC.
> When this option is ``None`` then UTC will be used.

Reviewer: I value the clarification but I still have questions (I think I now the answers, but I fear other readers may be confused):

1. UTC will be used and the returned timezones will be aware or not?
2. `this option` refers to `USE_TZ` or `TIME_ZONE`?

So I would suggest instead:

```suggestion
When :setting:`USE_TZ` is ``True``, reading datetimes from the database
returns aware datetimes.  If this option is set, the returned datetimes will
be in this time zone, otherwise UTC will be used.
```

Author: 1. Yep with `USE_TZ` then returned timestamps will be aware 
2. `TIME_ZONE`

The suggestion is ok but I'm not sure I think it's "better" 🤔 because it brings the "aware" part to the front which is secondary.

I guess the challenge here is explaining this pseudocode in natural language:

```
if USE_TZ = False:
   error!

else:
   timezone = DATABASES.TIME_ZONE
   if timezone is None:
        timezone = 'UTC'
```

The fact that timestamps are "aware" is secondary/implied if that makes sense? 🤔 

If you think that the suggestion makes that ^ pseudocode clear then I'm ok.

Reviewer: Thank you! I gotta confess, as I web service developer, I think I first try to confirm whether timestamps will be aware or not, and then, if "yes", I seek information about what timezones they will be in. But, OTOH, the first question should be answered (or seek answers in) the details of the `USE_TZ` setting, so I can see where you are coming from :-).

How about:

```suggestion
When :setting:`USE_TZ` is ``True``, reading datetimes from the database
returns aware datetimes with the timezone set to the value specified in
this option if not ``None``, or to ``UTC`` otherwise.
```

    third-party systems connect to the same database and expect to find
    datetimes in local time, then you must set this option.

> * If the database backend supports time zones (e.g., PostgreSQL), then the

Reviewer: I personally think that this comma does belong here, though I see that in some other places this pattern is used. Though from 981 "e.g." in the `django` folder, 41 are followed by a comma, and from 457 in the `docs` folder, 84 have the following comma.

Is this an English [from a specific region?] style pattern that I'm missing?

```suggestion
* If the database backend supports time zones (e.g. PostgreSQL), then the
```

Author: I don't see that pattern but it could just be my country🤷‍♂️

> * If the database backend supports time zones (e.g. PostgreSQL) then the
>   database connection's time zone set to this value.

Reviewer 2: ```suggestion
* If the database backend supports time zones (e.g., PostgreSQL), then the
  database connection's time zone is set to this value.
```

TIL.
e.g. meaning for example is always followed by a comma.

Author: Thanks missed a couple of other bits too 👍

>   Although setting the ``TIME_ZONE`` option is very rarely needed it may be
>   necessary to set this to the same value as the general :setting:`TIME_ZONE`
>   setting when raw queries that involve date/time functions such as
>   PostgreSQL's ``date_trunc()`` or ``generate_series()`` when generating
>   time-based series that transition daylight savings. It can be changed at any
>   time; the database takes care of converting datetimes to the desired time
>   zone.

Reviewer: This fragment seems like it has a grammatical issue: "_when raw queries that involve date/time functions ... when generating time-based series that transition daylight savings_"

Did you mean "_when raw queries that involve date/time functions ... generate time-based series that transition daylight savings_"?

What about:
```suggestion
  Although setting the ``TIME_ZONE`` option is very rarely needed, there are
  situations where it becomes necessary. Specifically, it should match the
  general :setting:`TIME_ZONE` setting when dealing with raw queries involving
  date/time functions like PostgreSQL's ``date_trunc()`` or
  ``generate_series()``, especially when generating time-based series that
  transition daylight savings.

  This value can be changed it at any time, the database will handle
  the conversion of datetimes to the configured time zone.
```

Author: lol I just re-read my sentence and it definitely is a challenge because it's too long and has "… when … when …" which is bad.

I like how you split it up 👍 

Re "Specifically, it should match … " I tried to avoid "should" style because it's not a hard rule.  I was going for "may be necessary" or "recommended". Otherwise your suggestion is perfect!

Reviewer: Ok! So, how about:

```suggestion
  Although setting the ``TIME_ZONE`` option is very rarely needed, there are
  situations where it becomes necessary. Specifically, it's recommended to
  match the general :setting:`TIME_ZONE` setting when dealing with raw queries
  involving date/time functions like PostgreSQL's ``date_trunc()`` or
  ``generate_series()``, especially when generating time-based series that
  transition daylight savings.

  This value can be changed it at any time, the database will handle
  the conversion of datetimes to the configured time zone.
```
